### PR TITLE
feat(igxPivot): Add superCompactMode Input for the grid that applies …

### LINKED ISSUE #10840: [Pivot] Add supercompactmode option for the pivot grid bound to class on a root level

### PR DESCRIPTION
**Pull request by MayaKirova**
…root level class.

Closes #10840   

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by MayaKirova** [comment]
@desig9stein Let me know if you think the class name is ok and whether you need any other classes applied anywhere else in the grid or if just one on the root is enough.

File: projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts [hunk 1]
```diff
@@ -151,6 +149,20 @@ export class IgxPivotGridComponent extends IgxGridBaseDirective implements OnIni
...
+    /**
+     * Enables a super compact theme for the component.
+     * @remarks
+     * Overrides the displayDensity option is one is set.
+     * @example
+     * ```html
+     * <igx-pivot-grid [superCompactMode]="true"></igx-pivot-grid>
+     * ```
+     */
+    @HostBinding('class.igx-grid__pivot---superCompact')
```
Comment (desig9stein) [inline projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts:162]:
```suggestion
@HostBinding('class.igx-grid__pivot--super-compact')
```

**Comment by dkamburov** [comment]
@desig9stein Is this one ready for testing?

**Review by simeonoff — APPROVED** [review]

**Comment by 3phase** [comment]
Using the superCompactMode, the columns and the values headers are intersecting, which breaks the layout.

image

**Comment by 3phase** [comment]
The second thing is that the chips for the filters, columns and values are not sized accordingly and that doesn't look good to me at all.
image

In contrast, when setting the display density to compact, the same chips look this way:
image

**Comment by 3phase** [comment]
@desig9stein, can you please look into the comments above?

**Comment by desig9stein** [comment]
@3phase  I fix the issue with the headers. @MayaKirova the problem with the chip comes from the density, I believe that we agree that in super-compact mode the density of the grid will always be set to compact, otherwise we have to override an enormous amount of styles.

**Comment by MayaKirova** [comment]
@desig9stein I was under the impression that all overrides will happen in the styles. One public property (superCompactMode) overriding another one (displayDensity) is not ideal.

**Pull request merged**